libraries/cmake/source/librdkafka/CMakeLists.txt

```diff
@@ -117,6 +117,15 @@ function(generateLibrdkafka)
       thirdparty_zlib
       thirdparty_zstd
 	)
+    message ( "${CMAKE_CURRENT_SOURCE_DIR}" )
+	configure_file(
+	"${CMAKE_CURRENT_SOURCE_DIR}/../zstd/src/lib/common/zstd_errors.h"
+	"${CMAKE_CURRENT_BINARY_DIR}/include/librdkafka/zstd_errors.h"
+	COPYONLY
+	)
+    target_include_directories(thirdparty_librdkafka_c PRIVATE
```

Comment:
@Smjert this seems to inform cmake about the default SYSTEM include dir
how can i call it from zstd and publish it since its a rdkafka source ?
i mean if i do the same target_include_directories(zstd_example PRIVATE will it notify also thirdparty_librdkafka_c about the Default SYSTEM include dir`s ?
i have also tried with SYSTEM DEFINITIONS but no luck...

Reply:
Take a look at this https://github.com/osquery/osquery/blob/6481b34e2382ffb5b4c469de3157c63e0ed23ea3/libraries/cmake/source/libelfin/CMakeLists.txt#L83-L108
Basically in CMake when you use the `INTERFACE` "visibility", it means that anything that links to that target will also inherit whatever is listed under INTERFACE.
So in your case, you move the configure_file into the libzstd CMake file, do there the `target_include_directories` with `SYSTEM INTERFACE`, as the example I linked, and it's done, since librdkafka already links `thirdparty_zstd`.

Reply:
Integrated in libztd